README.md

```diff
@@ -36,7 +36,7 @@ When deciding what to boot/update, the mcuboot bootloader looks at an installed
 
 By default, this header is configured to be 4kB in size. This can be adjusted using the configuration parameter `mcuboot.header_size`. 
 
-**However,** due to the way the FlashIAP block device currently works while erasing, the header_size should be configured to be the size of an erase sector (4kB in the case of an nRF52840). Erasing using the FlashIAPBlockDevice only works if the given address is erase-sector aligned!
+**However,** due to the way the FlashIAP block device currently works while erasing, the header_size should be configured to be the size of an erase sector (4kB in the case of an nRF52840). Erasing using the FlashIAPBlockDevice only works if the given address is erase-sector aligned! On the other hand, there is a hard upper limit of `< 65536` enforced by imgtool script. This needs to be carefully taken into consideration when choosing a MCU. Because a erase-sector of size >=4KB <64KB may actually not be available between the end of bootloader and start of primary application. For example, STM32F767/9xGx series MCU with dual bank enabled.
```

Comment:
Oof, that's not good.  Some STM32 MCUs have a 128k sector size!  Do we think it's worth trying to patch imgtool to increase the allowed size?

Reply:
I think that will help, at least let imgtool allow 64KB header. For the STM32F767/9 series MCUs with 1MB flash that I am working on right now, using 64KB header seems the best way to maximize space for primary application. The map will go like this: 64KB for bootloader, 64KB for header, the last 128KB sector for scratch. That leaves 0xC0000 for primary application and TLV trailers.

As of mcuboot v1.8.0, if I try to use 64KB hader size, imgtool will complain and quit:
```
struct.error: 'H' format requires 0 <= number <= 65535
```
But I don't know Python well enough to patch it.

Reply:
oooof, that sounds like it's trying to encode a struct for the header that only has a 16 bit size field.  I think this is worth raising as an issue to MCUBoot, to see if they have a recommended solution for this problem.  Maybe there's some way to make the header not need its own flash sector?  Surely someone has run into this issue before...

Reply:
And some STM32MCUs even have 256KB sectors such as STM32F767/9 that I am working on right now. Turning on dual bank mode will split those 256KB sectors into 128KB ones, but that then creates a weird map with a bunch of smaller(16/32KB) sectors sandwiched between a 128KB ones. But it is what it is.

Reply:
> oooof, that sounds like it's trying to encode a struct for the header that only has a 16 bit size field. I think this is worth raising as an issue to MCUBoot, to see if they have a recommended solution for this problem. Maybe there's some way to make the header not need its own flash sector? Surely someone has run into this issue before...

I am not sure if this is problem that affects all mcuboot ports(such as Zephyr etc.) or just mbed-os itself. From reading the doc written by Mr. Beckstein, I tend to believe this only affects mbed-os. Because he used IAP that needs to wide out an entire erase-sector before writing.

My understanding is that mcuboot is mainly backed by Nordic, and they use 4KB sectors in most/a lot of their MCUs, so I am not sure if they are interested in supporting STM. But we can try.

Reply:
Could you open an issue to ask?